runtime/src/lib.rs

```diff
@@ -572,6 +575,85 @@ impl_runtime_apis! {
 			System::account_nonce(account)
 		}
 	}
+	impl moonbeam_rpc_primitives_debug::DebugRuntimeApi<Block> for Runtime {
+		fn trace_transaction(
+			extrinsics: Vec<<Block as BlockT>::Extrinsic>,
+			transaction: &EthereumTransaction,
```

Comment:
The long body of this implementation should probably not live directly in the runtime. Maybe there should be a helper function or pallet for it?

Correct me if I'm wrong, but it should be relatively simple for other Frontier nodes to re-use this work right? And if this lives directly in the Runtime, it would need to be copy-pasted.

Reply:
By now I think we are OK(@crystalin ?). Please feel free to open a ticket for it and assign it to me if you think is a must.